modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -24,15 +24,18 @@ import * as InsertList from './InsertList';
 
 const wrappedElements = [ 'pre' ];
 
-const shouldPasteContentOnly = (fragment: AstNode, parentNode: Element) => {
-  const firstNode = fragment.firstChild;
-
-  const isAFlattenableTag = Arr.contains(wrappedElements, firstNode.name);
-  const isPastingInTheSameTag = firstNode.name === parentNode.tagName.toLowerCase();
-  const lastNode = fragment.lastChild.attr('data-mce-type') === 'bookmark' ? fragment.lastChild.prev : fragment.lastChild;
-  const isCopingOnlyOneTag = firstNode === lastNode;
-
-  return isCopingOnlyOneTag && isAFlattenableTag && isPastingInTheSameTag;
+const shouldPasteContentOnly = (dom: DOMUtils, fragment: AstNode, parentNode: Element, editor: Editor): boolean => {
+  const firstNode = fragment.firstChild as AstNode;
+  const lastNode = fragment.lastChild as AstNode;
+  const last = lastNode.attr('data-mce-type') === 'bookmark' ? lastNode.prev : lastNode;
+
+  const isCopyingOnlyOneTag = firstNode === last;
+  const isWrappedElement = Arr.contains(wrappedElements, firstNode.name);
+  const isPastingInTheSameTag = firstNode.name === parentNode.nodeName.toLowerCase() ? true : dom.getParent(parentNode, (node) => firstNode.name === node.nodeName.toLowerCase()) !== null;
```

Comment:
This can be simplified like so, because `getParent` runs on the node passed into it (it's weird but it's how it works):
```suggestion
  const isPastingInTheSameTag = dom.getParent(parentNode, (node) => firstNode.name === node.nodeName.toLowerCase()) !== null;
```

In saying that though, I have to wonder if this is really the correct logic we want. While this is okay for this case, since we only have one `wrappedElements` item, it wouldn't work well if we had multiple. For example, say the wrapped elements list had `pre` and `blockquote` in it. If we then pasted in a `blockquote` this would unwrap that and break things...

As such, I think a better solution would be to find the nearest block element and see if it compares as we really should only be unwrapping block elements. For example:
```ts
  const isPastingInTheSameBlockTag = dom.getParent(parentNode, dom.isBlock)?.nodeName == firstNode.name;
```

Reply:
Looks good, pretty much what I was trying to do with finding the parent block in the case `parentNode` is a `<strong>` or similar. Is the issue you mention with unwrapping when pasting `blockquote` content into another `blockquote` block? Or pasting into a different block also in `wrappedElements`?

Reply:
I'll give some demos shortly to explain it better as to be honest I don't think I've thought it through fully either, it was just more a potential code smell... what I've mentioned may not end up being an actual issue.

Reply:
Okay, so here's a slightly different case for if `li` was a wrapped element. So say we were pasting into this content:
```html
<ul>
<li>
  <p>
    insert here
  </p>
</li>
</ul>
```
and this was what was on the clipboard:
```html
<li>pasted content</li>
```
In that case, we wouldn't want this unwrapped and instead it should split the paragraph and start a new list. In saying that though, you shouldn't ever have just an `li` on the clipboard so it's still very much theoretical... either way, I'm happy to leave it with the version you have and we can adjust it later should we ever need to.